elasticapm/contrib/serverless/aws.py

```diff
@@ -50,11 +51,67 @@
 logger = get_logger("elasticapm.serverless")
 
 COLD_START = True
+INSTRUMENTED = False
 
-_AnnotatedFunctionT = TypeVar("_AnnotatedFunctionT")
+_AWSLambdaContextT = TypeVar("_AWSLambdaContextT")
 
 
-class capture_serverless(object):
+def capture_serverless(func: Optional[callable] = None, **kwargs) -> callable:
+    if not func:
+        # This allows for `@capture_serverless()` in addition to
+        # `@capture_serverless` decorator usage
+        return functools.partial(capture_serverless, **kwargs)
+
+    if kwargs:
+        warnings.warn(
+            PendingDeprecationWarning(
+                "Passing keyword arguments to capture_serverless will be deprecated in the next major release."
+            )
+        )
+
+    name = kwargs.pop("name", None)
+
+    # Disable all background threads except for transport
+    kwargs["metrics_interval"] = "0ms"
+    kwargs["breakdown_metrics"] = False
+    if "metrics_sets" not in kwargs and "ELASTIC_APM_METRICS_SETS" not in os.environ:
+        # Allow users to override metrics sets
+        kwargs["metrics_sets"] = []
+    kwargs["central_config"] = False
+    kwargs["cloud_provider"] = "none"
+    kwargs["framework_name"] = "AWS Lambda"
+    if "service_name" not in kwargs and "ELASTIC_APM_SERVICE_NAME" not in os.environ:
+        kwargs["service_name"] = os.environ["AWS_LAMBDA_FUNCTION_NAME"]
+    if "service_version" not in kwargs and "ELASTIC_APM_SERVICE_VERSION" not in os.environ:
+        kwargs["service_version"] = os.environ.get("AWS_LAMBDA_FUNCTION_VERSION")
+
+    global INSTRUMENTED
+    client = get_client()
+    if not client:
+        client = Client(**kwargs)
+    if not client.config.debug and client.config.instrument and client.config.enabled and not INSTRUMENTED:
+        elasticapm.instrument()
+        INSTRUMENTED = True
+
+    @functools.wraps(func)
+    def decorated(*args, **kwds):
+        if len(args) == 2:
+            # Saving these for request context later
+            event, context = args
+        else:
+            event, context = {}, {}
+
+        if not client.config.debug and client.config.instrument and client.config.enabled:
+            with capture_serverless_context(func, name, client, event, context) as sls:
+                sls.response = func(*args, **kwds)
```

Comment:
I wonder if it would make sense to move the code from `__exit__()` that accesses `response` to here, which would avoid us having to store the response on the `sls` object.

Reply:
I think I'm going to leave it. While I agree that it makes the context manager slightly more inscrutable to use, the reality is that the context manager is not designed to be used anywhere else, and I love how clean the `decorated()` function is.

I considered actually not using a context manager at all -- it's not really needed, as it's only used in one place. But I think the readability that it adds is worth it.